sdk/include/reactos/imm32_undoc.h

```diff
@@ -11,6 +11,17 @@
 #define KBDLAYOUT_IME 0xE000
 #define IS_IME_KBDLAYOUT(hKL) ((HIWORD(hKL) & KBDLAYOUT_MASK) == KBDLAYOUT_IME)
 
+/* unconfirmed */
+typedef struct tagCLIENTIMC {
+    HIMC   hImc;
```

Comment:
https://www.geoffchappell.com/studies/windows/win32/user32/structs/handleentry.htm lists 0x11 as being an IMC handle, so you might want to create handles like that?
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&sr=1&s=TYPE_INPUTCONTEXT

Reply:
See also
https://reactos.org/wiki/Techwiki:Win32k/IMC
https://reactos.org/wiki/Techwiki:Win32k/HANDLEENTRY

Reply:
Difficult to analyse, it takes time.